crates/common/tedge_utils/src/notify.rs

```diff
@@ -57,59 +61,79 @@ impl NotifyStream {
 
         // Automatically select the best implementation for your platform.
         // You can also access each implementation directly e.g. INotifyWatcher.
-        let watcher = RecommendedWatcher::new(
-            move |res: Result<notify::Event, notify::Error>| {
+        let watcher = debouncer::new_debouncer_opt(
+            NOTIFY_STREAM_DEBOUNCE_DURATION,
+            None,
+            move |res: Result<Vec<DebouncedEvent>, Vec<notify::Error>>| {
                 futures::executor::block_on(async {
-                    if let Ok(notify_event) = res {
-                        let event = match notify_event.kind {
-                            EventKind::Access(access_kind) => match access_kind {
-                                AccessKind::Close(access_mode) => match access_mode {
-                                    AccessMode::Any | AccessMode::Other | AccessMode::Write => {
-                                        Some(FsEvent::Modified)
+                    let Ok(debounced_events) = res else {
+                        return;
+                    };
+
+                    // simplify event type
+                    let mut debounced_events = debounced_events
+                        .into_iter()
+                        .filter_map(|notify_event| {
+                            let event = match notify_event.kind {
+                                EventKind::Access(access_kind) => match access_kind {
+                                    AccessKind::Close(access_mode) => match access_mode {
+                                        AccessMode::Any | AccessMode::Other | AccessMode::Write => {
+                                            Some(FsEvent::Modified)
+                                        }
```

Comment:
This is the branch that allows us to only listen to `FsEvent::Modified`. The actual sequence returned by lower-level `notify-debouncer-full` is:

```
[crates/common/tedge_utils/src/notify.rs:77] &notify_event = DebouncedEvent {
    event: Event {
        kind: Create(
            File,
        ),
        paths: [
            "/etc/tedge/plugins/tedge-log-plugin.toml",
        ],
        attr:tracker: None,
        attr:flag: None,
        attr:info: None,
        attr:source: None,
    },
    time: Instant {
        tv_sec: 23846,
        tv_nsec: 112231843,
    },
}
[crates/common/tedge_utils/src/notify.rs:77] &notify_event = DebouncedEvent {
    event: Event {
        kind: Access(
            Close(
                Write,
            ),
        ),
        paths: [
            "/etc/tedge/plugins/tedge-log-plugin.toml",
        ],
        attr:tracker: None,
        attr:flag: None,
        attr:info: None,
        attr:source: None,
    },
    time: Instant {
        tv_sec: 23846,
        tv_nsec: 112263954,
    },
}
[crates/common/tedge_utils/src/notify.rs:77] &notify_event = DebouncedEvent {
    event: Event {
        kind: Remove(
            File,
        ),
        paths: [
            "/etc/tedge/plugins/tedge-log-plugin.toml~",
        ],
        attr:tracker: None,
        attr:flag: None,
        attr:info: None,
        attr:source: None,
    },
    time: Instant {
        tv_sec: 23846,
        tv_nsec: 112322524,
    },
}
```

when saving the file with nvim, or:

```
[crates/common/tedge_utils/src/notify.rs:77] &notify_event = DebouncedEvent {
    event: Event {
        kind: Create(
            File,
        ),
        paths: [
            "/etc/tedge/plugins/tedge-log-plugin.toml",
        ],
        attr:tracker: None,
        attr:flag: None,
        attr:info: None,
        attr:source: None,
    },
    time: Instant {
        tv_sec: 23953,
        tv_nsec: 87955441,
    },
}
[crates/common/tedge_utils/src/notify.rs:77] &notify_event = DebouncedEvent {
    event: Event {
        kind: Access(
            Close(
                Write,
            ),
        ),
        paths: [
            "/etc/tedge/plugins/tedge-log-plugin.toml",
        ],
        attr:tracker: None,
        attr:flag: None,
        attr:info: None,
        attr:source: None,
    },
    time: Instant {
        tv_sec: 23953,
        tv_nsec: 87977201,
    },
}
```

when file is copied. Picking up `AccessKind::Close(AccessMode::Write)` allows us to reload appropriately.